server/integration-tests/src/test/java/io/smallrye/graphql/tests/client/typesafe/uni/TypesafeClientUniTest.java

```diff
@@ -0,0 +1,79 @@
+package io.smallrye.graphql.tests.client.typesafe.uni;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URL;
+import java.time.Duration;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.smallrye.graphql.client.typesafe.vertx.VertxTypesafeGraphQLClientBuilder;
+
+/**
+ * Test for usage of Uni in typesafe clients.
+ * Generally, to the client, it should not matter at all whether the server-side query returns Uni or a synchronous type.
+ * In either case, the client-side counterpart of the method can return a synchronous or an asynchronous type.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class TypesafeClientUniTest {
+
+    @Deployment
+    public static WebArchive deployment() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addClasses(UniApi.class);
+    }
+
+    @ArquillianResource
+    URL testingURL;
+
+    private static UniClientApi client;
```

Comment:
```suggestion
    private UniClientApi client;
```
... or do you mean the prepare/cleanup methods to also be static and @BeforeClass/@AfterClass?